DatadogFlags/Tests/FlagsClientNamedInstanceTests.swift

```diff
@@ -0,0 +1,156 @@
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import XCTest
+import TestUtilities
+import DatadogInternal
+@testable import DatadogFlags
+
+final class FlagsClientNamedInstanceTests: XCTestCase {
+    override func setUp() {
+        super.setUp()
+        // Clean up registry before each test
+        for instanceName in FlagsClientRegistry.registeredInstanceNames() {
+            FlagsClientRegistry.unregisterInstance(named: instanceName)
+        }
+    }
+
+    override func tearDown() {
+        // Clean up registry after each test
+        for instanceName in FlagsClientRegistry.registeredInstanceNames() {
+            FlagsClientRegistry.unregisterInstance(named: instanceName)
+        }
+        super.tearDown()
+    }
+
+    func testCreateWithNameOnly() {
+        let core = FeatureRegistrationCoreMock()
+        Flags.enable(in: core)
+
+        let client = FlagsClient.create(name: "test-client", in: core)
+        XCTAssertFalse(client is NOPFlagsClient, "Should create a functional client")
+
+        XCTAssertTrue(FlagsClientRegistry.isRegistered(instanceName: "test-client"))
+
+        let retrieved = FlagsClientRegistry.instance(named: "test-client")
+        XCTAssertTrue(retrieved === client, "Should register and retrieve the same instance")
+    }
+
+    func testCreateWithConfiguration() {
+        let core = FeatureRegistrationCoreMock()
+        Flags.enable(in: core)
+
+        let config = FlagsClient.Configuration(
+            baseURL: "https://custom.endpoint.com",
+            customHeaders: ["X-Custom": "value"],
+            flaggingProxy: "proxy.example.com"
+        )
+
+        let client = FlagsClient.create(with: config, name: "configured-client", in: core)
+        XCTAssertFalse(client is NOPFlagsClient, "Should create a functional client")
+
+        XCTAssertTrue(FlagsClientRegistry.isRegistered(instanceName: "configured-client"))
+
+        let retrieved = FlagsClientRegistry.instance(named: "configured-client")
+        XCTAssertTrue(retrieved === client, "Should register and retrieve the same instance")
+    }
+
+    func testInstanceRetrieval() {
+        let core = FeatureRegistrationCoreMock()
+        Flags.enable(in: core)
+
+        // Create multiple named instances
+        let client1 = FlagsClient.create(name: "client1", in: core)
+        let client2 = FlagsClient.create(name: "client2", in: core)
+        let client3 = FlagsClient.create(name: "client3", in: core)
+
+        // Verify each can be retrieved correctly
+        XCTAssertTrue(FlagsClient.instance(named: "client1") === client1)
+        XCTAssertTrue(FlagsClient.instance(named: "client2") === client2)
+        XCTAssertTrue(FlagsClient.instance(named: "client3") === client3)
+
+        // Non-existent instance should return NOP
+        let nonExistent = FlagsClient.instance(named: "nonexistent")
+        XCTAssertTrue(nonExistent is NOPFlagsClient)
```

Comment:
`nonExistent` is eligible for `===` comparisons, but we don't have a way to check equality with another `NOPFlagsClient` instance because if we generate one, it'll be a different instance

One option I thought about was to do something like `XCTAssertTrue(nonExistent === NOPFlagsClient.shared)` where `NOPFlagsClient.shared` would reference a singleton NOP instance. But that breaks the expectation that `FlagsClient.create` and `FlagsClient.instance` don't return singletons, although one could argue that named clients are singletons within the scope of their name